packages/wp-story-editor/src/components/mediaUpload/mediaPicker/useMediaPicker.js

```diff
@@ -92,7 +99,35 @@ function useMediaPicker({
     } catch (e) {
       // Silence.
     }
-  }, [updateMedia]);
+  }, [onUpload, updateMedia]);
+
+  useEffect(() => {
+    const currentDetails = window.wp.media.view.Attachment.Details;
+    const wsDetails = currentDetails?.extend({
+      deleteAttachment: function () {
+        currentDetails.prototype.deleteAttachment.apply(
+          this,
+          // eslint-disable-next-line prefer-rest-params -- expected.
+          arguments
+        );
+        if (onDelete) {
+          onDelete(getResourceFromMediaPicker(this.model.attributes));
+        }
+      },
+      trashAttachment: function () {
+        currentDetails.prototype.trashAttachment.apply(
+          this,
+          // eslint-disable-next-line prefer-rest-params -- expected.
+          arguments
+        );
+
+        if (onDelete) {
+          onDelete(getResourceFromMediaPicker(this.model.attributes));
+        }
+      },
+    });
+    window.wp.media.view.Attachment.Details = wsDetails;
```

Comment:
Probably want to do a cleanup here:

```suggestion
    window.wp.media.view.Attachment.Details = wsDetails;
    
    return () => {
      window.wp.media.view.Attachment.Details = currentDetails;
    };
```